### DISCUSSION TITLE
Добавление цикла в таблицу циклов кнопкой из временной таблицы

### DISCUSSION THREAD

**Discussion by an220022** [totumonline/totum-ru-issues-and-discussions#34] [Как это сделать?] [answered]
Есть временная таблица, которая выводит определенные строки в зависимости от значений хедера. В строку добавлена кнопка, которая может передать текущие значения хедера и значения строки для создания нового цикла. Каким кодом кнопки можно инициализировать создание цикла с передачей параметров из текущей таблицы?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### [Видео на YouTube —>](https://youtu.be/RX-EpctWUb0)

#### Общий подход такой:

1. Создается цикла через `insert` в таблицу циклов с записью добавленных строк в переменную (параметр `inserts` функции `insert`).

2. В этом же коде следующим действием делаем `set` с использованием в параметр `cycle` данных из переменной:

```

a1= : insert(table: 's2_cycles'; inserts: "inserts"; log: true)

a2=: set(table: 's2_cycle'; cycle: $#inserts[0]; field: 'h_field_a' = #h_field_a; field: 'h_field_b' = #h_field_b; field: 'h_field_c' = #h_field_c;  log: true)


```

#### Также этот подход можно использовать для того, что бы заменить стандартную кнопку добавления цикла.

1. Создаем в таблице циклов кастомную кнопку добавления вызывающую временную таблицу в которой есть описанное выше создание цикла.

2. Подключаем эту кнопку в кнопки строчной части параметром `buttons` функции `setTableFormat`.

3. Отключаем стандартную кнопку добавления убрав у таблицы циклов параметр `Добавление`.
